aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/SigV4SigningDecorator.kt

```diff
@@ -111,23 +112,29 @@ class SigV4SigningConfig(
                     *codegenScope
                 )
                 if (serviceHasEventStream) {
-                    rustTemplate(
-                        """
-                        /// Creates a new Event Stream `SignMessage` implementor.
-                        pub fn new_event_stream_signer(
-                            &self,
-                            properties: #{SharedPropertyBag}
-                        ) -> #{SigV4Signer} {
-                            #{SigV4Signer}::new(properties)
-                        }
-                        """,
-                        *codegenScope
-                    )
+                    inner()
                 }
             }
             else -> emptySection
         }
     }
+
+    override fun inner(): Writable {
+        return writable {
+            rustTemplate(
+                """
+                /// Creates a new Event Stream `SignMessage` implementor.
+                pub fn new_event_stream_sigv4_signer(
```

Comment:
Should we make this `pub(crate)`? I can't think of a reason a customer would need to call this.

Separately, would it be possible to have one `new_event_stream_signer` function that returns `impl SignMessage`?

Reply:
It is possible, but I'd either make `MergedEventStreamDecorator` return the first that applies or take only one `EventStreamDecorator`; we can't have multiple `fn new_event_stream_signer` anyway and we'd have to be careful while coding (so maybe better to prevent it?)

Reply:
Is there a good reason for having multiple?

Reply:
I thought we could generate all that apply, but then I'll remove the list and use only the relevant one and keep `applies()` as it was before